packages/flutter_test/lib/src/matchers.dart

```diff
@@ -2138,6 +2150,39 @@ class _CoversSameAreaAs extends Matcher {
     description.add('covers expected area and only expected area');
 }
 
+class _ColorSwatchMatcher<T> extends Matcher {
+  _ColorSwatchMatcher(this._target, this._threshold);
+
+  final ColorSwatch<T> _target;
+  final double _threshold;
+
+  @override
+  Description describe(Description description) {
+    return description.add('matches color swatch "$_target" with threshold "$_threshold".');
+  }
+
+  @override
+  bool matches(dynamic item, Map<dynamic, dynamic> matchState) {
+    if (item is ColorSwatch) {
+      final _ColorMatcher matcher = _ColorMatcher(_target, _threshold);
+      if (!matcher.matches(item, matchState)) {
+        return false;
+      }
+
+      for (final T key in _target.keys) {
```

Comment:
We should check that keys.length is the same for _target and item; otherwise one would be a subset of the other.

Reply:
done